src/main/java/com/vmware/vipclient/i18n/base/cache/CacheItem.java

```diff
@@ -4,6 +4,51 @@
  */
 package com.vmware.vipclient.i18n.base.cache;
 
-public interface CacheItem {
-    public boolean isExpired();
+import com.vmware.vipclient.i18n.VIPCfg;
+
+public abstract class CacheItem {
+    private String etag;
+    private long timestamp;
+    private Long maxAgeMillis = 86400000l;
+
+    public String getEtag() {
+        return etag;
+    }
+
+    protected void setEtag(String etag) {
+        this.etag = etag;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    protected void setTimestamp(long timestamp) {
```

Comment:
remove this or make this synchronized ?

Reply:
Same comment: it has been like this before, the only change in this PR is to put common code inside the common CacheItem instead of implementing twice in MessageCacheItem and FormatCacheItem. Let's make any code logic change separately.

Also, only the setters inside child classes (MessageCacheItem, PatternCacheItem, LocaleCAcheItem) use this protected setTimestamp.The setters are synchronized and have the following signature:

`public synchronized void setCacheItem(String etag, long timestamp, Long maxAgeMillis) `

Reply:
Why unnecessary? The 3 properties: etag, timestamp and maxAgeMillis have to be set in a single thread at the same time to avoid any thread collision issue.

As I said, there isn't any code logic change here in this PR. Thread-safety implementation has has been like this before this PR, so could you log a separate issue regarding any thread-safety problems you are seeing?